sdk/samples/iot/paho_iot_hub_twin_cbor_sample.c

```diff
@@ -0,0 +1,533 @@
+
+#define SAMPLE_TYPE PAHO_IOT_HUB
+#define SAMPLE_NAME PAHO_IOT_HUB_TWIN_SAMPLE
+
+#define MAX_TWIN_MESSAGE_COUNT 5
+#define MQTT_TIMEOUT_RECEIVE_MS (60 * 1000)
+#define MQTT_TIMEOUT_DISCONNECT_MS (10 * 1000)
+
+static az_span const twin_document_topic_request_id = AZ_SPAN_LITERAL_FROM_STR("get_twin");
+static az_span const twin_patch_topic_request_id = AZ_SPAN_LITERAL_FROM_STR("reported_prop");
+
+static char* const desired_device_count_property_name = "device_count";
+static int64_t device_count_value = 0;
+
+static iot_sample_environment_variables env_vars;
+static az_iot_hub_client hub_client;
+static MQTTClient mqtt_client;
+static char mqtt_client_username_buffer[256];
+
+// Functions
+static void create_and_configure_mqtt_client(void);
+static void connect_mqtt_client_to_iot_hub(void);
+static void subscribe_mqtt_client_to_iot_hub_topics(void);
+static void send_and_receive_device_twin_messages(void);
+static void disconnect_mqtt_client_from_iot_hub(void);
+
+static void get_device_twin_document(void);
+static void send_reported_property(void);
+static bool receive_device_twin_message(void);
+static void parse_device_twin_message(
+    char* topic,
+    int topic_len,
+    MQTTClient_message const* message,
+    az_iot_hub_client_twin_response* out_twin_response);
+static void handle_device_twin_message(
+    MQTTClient_message const* message,
+    az_iot_hub_client_twin_response const* twin_response);
+static bool parse_desired_device_count_property(
+    MQTTClient_message const* message,
+    int64_t* out_parsed_device_count);
+static void update_device_count_property(int64_t device_count);
+static void build_reported_property(
+    uint8_t* reported_property_payload,
+    size_t reported_property_payload_size,
+    size_t* out_reported_property_length);
+
+/*
+ * This sample utilizes the Azure IoT Hub to get the device twin document in CBOR, send a reported
+ * property message in CBOR, and receive up to 5 desired property messages in CBOR. If a timeout
+ * occurs while waiting for a message from the Azure IoT Hub, the sample will exit. Upon receiving a
+ * desired property message, the sample will update the twin property locally and send a reported
+ * property message back to the service. X509 self-certification is used.
+ *
+ * A property named `device_count` is supported for this sample. To send a device twin desired
+ * property message, select your device's Device Twin tab in the Azure Portal of your IoT Hub. Add
+ * the property `device_count` along with a corresponding value to the `desired` section of the
+ * JSON. Select Save to update the twin document and send the twin message to the device.
+ *
+ * {
+ *   "properties": {
+ *     "desired": {
+ *       "device_count": 42,
+ *     }
+ *   }
+ * }
+ *
+ * No other property names sent in a desired property message are supported. If any are sent, the
+ * IOT_SAMPLE_LOG will report there is nothing to update.
+ */
+int main(void)
+{
+  create_and_configure_mqtt_client();
```

Comment:
maybe have a print statement here which calls out that preview hubs are needed? (I think they're needed). And to expect a disconnect if you use this without one (if that's the behavior). sorry some ignorance here on what exactly happens between device and hub for this feature.

Reply:
This only works on a Dogfood hub right now (not an official public preview hub).  There is no official public preview right now of this feature.  Also, I'm concerned that if commenting is tied too closely to a feature branch, it will get missed when the merge into master happens, and then be wrong.  I'd like to bring up feature branch management and public-preview branch management in the Friday meeting.